operators/multiclusterobservability/controllers/placementrule/placementrule_controller.go

```diff
@@ -354,6 +354,11 @@ func createAllRelatedRes(
 					addonConfig,
 				)
 				if err != nil {
+					if k8serrors.IsNotFound(err) {
```

Comment:
AddOnDeploymentConfig resource will still be found right from Get? since it was not deleted? I think the check should be for if config.DefaultConfig is nil then empty out defaultAddonDeploymentConfig

Reply:
Just saw your one line change. I guess this will not apply :)

Reply:
Yep. It's much more simple to standard with the default case value (use an empty config). Only if we find a config we update the reference. This way we don't need to handle all the possible scenario in which the config should be set to empty.